lib/Lib.Buffer.fsti

```diff
@@ -79,21 +79,59 @@ let const_to_lbuffer #a #len (b:clbuffer a len{CB.qual_of (b <: cbuffer a) == CB
 let const_to_ilbuffer #a #len (b:glbuffer a len)  : r:ilbuffer a len =
   const_to_ibuffer #a b
 
-unfold let null (a: Type0) : lbuffer a (size 0) = B.null #a
+unfold let null (ty : buftype) (a : Type0) : buffer_t ty a =
+  match ty with
+  | IMMUT -> IB.inull #a
+  | MUT -> B.null #a
+  | CONST -> CB.null a
+
+let g_is_null (#ty : buftype) (#a : Type0) (b : buffer_t ty a) : GTot bool =
+  match ty with
+  | IMMUT -> IB.g_is_null (b <: ibuffer a)
+  | MUT -> B.g_is_null (b <: buffer a)
+  | CONST -> B.g_is_null (CB.as_mbuf (b <: cbuffer a))
 
-//val live: #t:buftype -> #a:Type0 -> h:HS.mem -> b:buffer_t t a -> Type
 let live (#t:buftype) (#a:Type0) (h:HS.mem) (b:buffer_t t a) : Type =
```

Comment:
Change this from `t` to `ty` everywhere if you are gonna use this

Reply:
I actually changed all the `ty` to `t` for the buffer type